src/xdocs/config_annotation.xml

```diff
@@ -228,6 +228,111 @@ public void test(&#64;MyAnnotation String s) { // OK
       </subsection>
     </section>
 
+    <section name="AnnotationOnSameLine">
+      <subsection name="Description">
+        <p>Since Checkstyle 8.2</p>
+        <p>
+          Check annotation located on same line with its target.
```

Comment:
We need make a note that this Check is not a good practice, but some style guides use it.

Reply:
Make sure this note is in the check's Javadoc.

Reply:
Note added to both xdocs and javadoc